test/with_dummyserver/test_connectionpool.py

```diff
@@ -1112,28 +1135,28 @@ def test_retry_reuse_safe(self):
             )
             assert resp.status == 200
 
-    def test_retry_return_in_response(self):
+    def test_retry_return_in_response(self) -> None:
         with HTTPConnectionPool(self.host, self.port) as pool:
             headers = {"test-name": "test_retry_return_in_response"}
             retry = Retry(total=2, status_forcelist=[418])
             resp = pool.request(
                 "GET", "/successful_retry", headers=headers, retries=retry
             )
             assert resp.status == 200
-            assert resp.retries.total == 1
-            assert resp.retries.history == (
+            assert resp.retries.total == 1  # type: ignore[attr-defined]
```

Comment:
Which attr is not defined here, and why?

Reply:
it was `retries` and the `request` method returned `BaseHTTPResponse`.
I've added a commit and changed the return type to `HTTPResponse`
Thanks for bringing this.

Reply:
We should add `retries` to `BaseHTTPResponse`.